controllers/providers/aws/ssm.go

```diff
@@ -0,0 +1,56 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
+	"github.com/keikoproj/aws-sdk-go-cache/cache"
+	"github.com/keikoproj/instance-manager/controllers/common"
+)
+
+const (
+	EksOptimisedAmiPath = "/aws/service/eks/optimized-ami/%s/amazon-linux-2/recommended/image_id"
+)
+
+func GetAwsSsmClient(region string, cacheCfg *cache.Config, maxRetries int, collector *common.MetricsCollector) ssmiface.SSMAPI {
+	config := aws.NewConfig().WithRegion(region).WithCredentialsChainVerboseErrors(true)
+	config = request.WithRetryer(config, NewRetryLogger(maxRetries, collector))
+	sess, err := session.NewSession(config)
+	if err != nil {
+		panic(err)
+	}
+	cache.AddCaching(sess, cacheCfg)
+	cacheCfg.SetCacheTTL("ssm", "GetParameter", GetParameterTTL)
+	sess.Handlers.Complete.PushFront(func(r *request.Request) {
+		ctx := r.HTTPRequest.Context()
+		log.V(1).Info("AWS API call",
+			"cacheHit", cache.IsCacheHit(ctx),
+			"service", r.ClientInfo.ServiceName,
+			"operation", r.Operation.Name,
+		)
+	})
+	return ssm.New(sess)
+}
+
+func (w *AwsWorker) GetEksLatestAmi(clusterName string) (string, error) {
```

Comment:
This function will need to handle variability across OS families and architectures - i.e. it should be able to get the latest AMIs for:

- Bottlerocket OS running x86_64
- Bottlerocket OS running ARM
- Amazon Linux 2 running x86_64
- Amazon Linux 2 running ARM
- Windows 2019 running x86_64

Larger organizations will also generally prefer the ability to customize this - i.e. have a custom SSM path to point to.

Reply:
All good points. Do you think they should be implemented in a v1 of this feature?
Or can we implement what's needed today and expand it to fit need?

Reply:
I'd prefer to implement as V1 - or an _extremely_ fast follow - at least for detection based off of os-family and architecture. I don't think custom paths is required for v1.

Reply:
Correct me if I'm wrong, but doesn't that information come _from_ the AMI? So we'd need to get the user to pass in somehow what architecture they want to use?

Did you have a suggestion on how that should work?

The simplest and ugliest way I can think of is to encode that information into the string; have instead of `latest`, `latest-al2-x86`, `latest-bottlerocket-arm64` or `latest-windows-x86`

Reply:
Not necessarily - the IG will be annotated with the os-family being used (defaults to al2). For architecture, it can be derived from the type of instance that is being used - i.e. M5 family will always be x86, M6g family will always be arm

Reply:
just found `instancemgr.keikoproj.io/os-family` ok, that should be easy - assuming amazon provide a latest ami for those variations :)

Reply:
I have to choose between Windows Cofe and Windows Full; any preferences?

Reply:
Core